File: 1542-consecutive-characters/consecutive-characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        if not s:
            return 0
        count  =  1
        maxCount = 1
        for i in range(1,len(s)):
            if s[i] == s[i - 1]:
                count += 1
            else:
                maxCount = max(count , maxCount)
                count = 1
        maxCount = max(count, maxCount)
        return maxCount

File: 1542-consecutive-characters/consecutive-characters.cpp
class Solution {
public:

    int maxPower(string s) {
        if (s.empty()) return 0; 
        int maxCount = 1;
        int count = 1;

        for (int i = 1; i < s.length(); i++) {
            if (s[i] == s[i - 1]) {
                count++;
            } else {
                maxCount = max(maxCount, count);
                count = 1; 
            }
        }

        maxCount = max(maxCount, count);

        return maxCount;
    }
};